### DISCUSSION TITLE
Can't send to gmail from DigitalOcean server

### DISCUSSION THREAD

**Discussion by gazugafan** [postalserver/postal#2076] [Help with using Postal] [answered]
Setup a new postal install on a fresh DigitalOcean droplet. Everything seems to be working great except when trying to send to a gmail.com address. Here's the activity on a gmail message...
![image](https://user-images.githubusercontent.com/854538/179321293-4be8d287-2621-4b40-9fad-e26cf35492d8.png)

And I see these in the worker logs for all the gmail-like SMTP connection attempts...
```
 Cannot connect to 172.253.115.27:25 (gmail-smtp-in.l.google.com) (Net::OpenTimeout: execution expired)
```
Pinging one of those domains seems to be fine...
```
ping alt4.gmail-smtp-in.l.google.com
PING alt4.gmail-smtp-in.l.google.com(ea-in-f27.1e100.net (2a00:1450:4013:c16::1b)) 56 data bytes
64 bytes from ea-in-f27.1e100.net (2a00:1450:4013:c16::1b): icmp_seq=1 ttl=105 time=77.9 ms
```
Trying to telnet to them on port 25 is more interesting...
```
telnet alt4.gmail-smtp-in.l.google.com 25
Trying 2a00:1450:4013:c16::1b...
***ABOUT A 1 MINUTE DELAY***
Trying 142.250.153.27...
Connected to alt4.gmail-smtp-in.l.google.com.
Escape character is '^]'.
220 mx.google.com ESMTP g14-20020a1709065d0e00b0071584915a2asi5681905ejt.602 - gsmtp
quit
221 2.0.0 closing connection g14-20020a1709065d0e00b0071584915a2asi5681905ejt.602 - gsmtp
Connection closed by foreign host.
```
It tries to use the IPv6 address first, but this hangs and times out after about 1 minute. Then it tries the IPv4 address and connects immediately. I'm wondering if Postal is only trying the IPv6 address and never falls back to IPv4?

I've also heard tales of Google blocking all the IPs in DigitalOcean's range. I suppose that would do it as well, but the telnet being able to connect on IPv4 seems to rule that out.

Thoughts?

**Comment by willpower232** [comment, marked as answer]
I believe DigitalOcean blocks emails being sent on newer accounts so you should double check with them first whether or not they're allowing you to send messages.

It is interesting that telnet eventually works so if DigitalOcean say they aren't responsible you can double check things like mxtoolbox.com or email-tester.com

**Reply by gazugafan** [reply]
Thanks! It seems that DigitalOcean is indeed blocking port 25, but only over IPv6, which was adding to the confusion. So I think the simplest solution here is to just disable IPv6 on the server. That should solve it. Will report back if anything else related comes up :)

**Reply by willpower232** [reply]
Haha definitely a shortcut but probably not great long term, it might be easier to migrate to a different server provider where you can freely send emails.

**Reply by gazugafan** [reply]
Yeah, it's not ideal! I'm going back and forth with DigitalOcean support now to try to get them to open port 25 on IPv6. No luck so far 😓. Any suggestions on a better server provider?

**Reply by gazugafan** [reply]
Doesn't sound like DigitalOcean support will budge with opening port 25 on IPv6. They say it's blocked at a "network level" for all customers, so I assume this means there's no way it can be opened on a per-account or droplet basis.
https://docs.digitalocean.com/products/networking/ipv6/details/limits

At least that answers the question! It might be worth making a note of this in the documentation here, since Digital Ocean is mentioned specifically...
https://docs.postalserver.io/install/prerequisites

The only solution to use Postal on DigitalOcean, it seems, is to create a droplet with IPv6 disabled.
